src/GrpcProtos/capture.proto

```diff
@@ -599,6 +599,28 @@ message ThreadStateSlice {
   // Otherwise, they should be 0 to minimize bandwidth usage.
   uint32 wakeup_tid = 7;
   uint32 wakeup_pid = 8;
+
+  // This is the callstack of the tracepoint that changed the state
+  // of the thread and caused this slice to happen. If the thread
+  // state is runnable then this callstack is coming from wakeup_tid,
```

Comment:
Ok, we need to be careful here as this is messy.

We have `wakeup_reason` that controls whether `wakeup_tid` is valid, but then we have the state itself that controls whether `triggering_callstack_id` is from this thread or `wakeup_tid`. This is explained here but is still awkward when using the proto. 

Would it correct to say that this callstack comes from this thread if `wakeup_reason` is `kNotApplicable`, and from `wakeup_tid` otherwise?

Reply:
Yeah, I think this is true.